static/js/util/redux.js

```diff
@@ -0,0 +1,25 @@
+// @flow
+import type {
+  ActionCreator,
+  AsyncActionHelper,
+  AsyncActionCreator
+} from '../flow/reduxTypes';
+import type { Dispatch } from 'redux';
+
+export function createActionHelper(dispatch: Dispatch, actionCreator: Function): (...args: any) => void {
+  return (...args) => dispatch(actionCreator(...args));
+}
+
+export type ActionHelpers = Array<{[k: string]: (...args: any) => void}>;
+export function createSimpleActionHelpers(dispatch: Dispatch, actionManifest: ActionCreator[]): ActionHelpers {
+  return actionManifest.map(actionCreator => (
+    { [actionCreator.name]: createActionHelper(dispatch, actionCreator) }
+  ));
+}
+
+export type AsyncActionHelpers = Array<{[k: string]: AsyncActionHelper}>;
+export function createAsyncActionHelpers(dispatch: Dispatch, actionManifest: AsyncActionCreator[]): AsyncActionHelpers {
+  return actionManifest.map(actionCreator => (
+    { [actionCreator.name]: createActionHelper(dispatch, actionCreator) }
+  ));
+}
```

Comment:
so specifying the return type actually has an affect on the functionality? i thought it was just a sort of 'wrapper' to validate types going in and out. i'm so confused - it looks like we have 2 blocks of the same exact javascript, but we're getting something different from each block because we've added flow typing. i'm willing to thumbs-up this, but i'd like to understand this first